pkg/agent/nodeportlocal/k8s/npl_controller.go

```diff
@@ -282,11 +289,13 @@ func (c *NPLController) isNPLEnabledForServiceOfPod(obj interface{}) bool {
 		if isSvc && svc.Spec.Type != corev1.ServiceTypeNodePort {
 			if pod.Namespace == svc.Namespace &&
 				matchSvcSelectorPodLabels(svc.Spec.Selector, pod.GetLabels()) {
-				return true
+				for _, port := range svc.Spec.Ports {
+					targetPorts.Insert(int(port.TargetPort.IntVal))
```

Comment:
`TargetPort` can be a string (named port in the target Pod's container). If we don't want to support this case yet, it's fine, but we should check for that case and log an error.

Reply:
but it seems that we should be able to support named ports here, by returning both a list of numerical ports and a list of named ports (strings), and letting `handleAddUpdatePod` do the resolution from port name to port number.